data/blog/intro-to-entity-framework-core-in-net-6.md

```diff
@@ -0,0 +1,586 @@
+images: []
+layout: PostLayout
+canonicalUrl: ''
+authors: ['djoufson']
+---
+
+# Intro to Entity Framework Core in .NET 6
+
+![Intro to Entity Framework Core in .NET 6](/static/images/posts/intro-to-entity-framework-core-in-net-6/banner.jpg)
+
+Entity framework is a .NET tool to interact with various kind of databases, providing the same api for each of these database providers. In this blog post, you will see how to set un Entity Framework and build your first CRUD operation over a Sqlite database
+
+# Overview
+
+- [Requirements](#requirements)
+- [Project setup](#project-setup)
+- [Create the entities](#create-the-entities)
+- [Connect to the database](#connect-to-the-database)
+- [Test over the database](#test-over-the-database)
+- [Next steps](#next-steps)
+
+# Requirements
+
+For this tutorial I'll be using Vs Code as my code editor, and all the stuff will be made through CLI. You can use your favorite Code editor, and your favorite CLI tool (Powershell, CMD, cmdr ...)
+
+## Install .NET
+
+Since this is a demo of efcore using .NET 6, you first need to download and install the [.NET 6 SDK](). Make sure you install the one compatible with your processor architecture.
+
+## Install efcore cli tool
+
+To install the CLI tool of Entity framework, you need to run the following commands
+
+```shell
+dotnet tool install --global dotnet-ef
+```
+
+```shell
+dotnet tool update --global dotnet-ef
+```
+
+After those two commands, you can run the next one to confirm that everything was installed successfully:
+
+```shell
+dotnet ef
+```
+
+And the output should be:
+![Test Ef core installation](/static/images/posts/intro-to-entity-framework-core-in-net-6/ef-test.png)
+
+# Project setup
+
+Once the required tools are installed, we can go ahead and create our C# project.
+
+To follow along with me, you should run the exact commands shown below. I will obviously explain why I run a specific command, and what is it for.
+
+## Create a Console project
+
+```shell
+dotnet new console -o "Ef-Intro"
+```
+
+```shell
+cd Ef-Intro
+```
+
+Those two commands are to create a blank .NET Console application. Open the `Ef-Intro` created folder in your code editor.
+
+To run the project, type:
+
+```shell
+dotnet run
+```
+
+`Output`
+
+```
+Hello, World!
+```
+
+## Install Entity Framework Packages
+
+Type the following commands in the root directory of your project
+
+```shell
+dotnet add package Microsoft.EntityFrameworkCore.Tools --version 7.0.5
+```
+
+```shell
+dotnet add package Microsoft.EntityFrameworkCore.Design --version 7.0.5
+```
+
+```shell
+dotnet add package Microsoft.EntityFrameworkCore --version 7.0.5
+```
+
+```shell
+dotnet add package Microsoft.EntityFrameworkCore.Sqlite.Core --version 7.0.5
```

Comment:
Is there a particular reason why you pin the version of entity core? If yes can you specify in the title of the blog post the version of efcore that you'd be using?